tests/unit/ssg-module/test_build_renumber.py

```diff
@@ -0,0 +1,385 @@
+    vars = ET.SubElement(oval, "{%s}variables" % oval_namespace)
+    e_var = ET.SubElement(vars, "{%s}external_variable" % oval_namespace)
+    e_var.set("id", "var_password_age")
+    return oval
+
+
+@pytest.fixture
+def oval_with_external_variable():
+    oval = ET.Element("{%s}oval_definitions" % oval_namespace)
+    vars = ET.SubElement(oval, "{%s}variables" % oval_namespace)
+    e_var = ET.SubElement(vars, "{%s}external_variable" % oval_namespace)
+    e_var.set("id", "var_password_age")
+    e_var.set("datatype", "string")
+    e_var.set("comment", "this is your long age")
+    e_var.set("version", "1")
+    return oval
+
+
+def test_check_and_correct_xccdf_to_oval_data_export_matching_constraints_1(
+        xccdf_with_value, oval_with_external_variable_without_id_or_datatype):
+    with pytest.raises(SSGError) as e:
+        ssg.build_renumber.check_and_correct_xccdf_to_oval_data_export_matching_constraints(
+            xccdf_with_value,
+            oval_with_external_variable_without_id_or_datatype)
+    assert (
+        "Invalid OVAL <external_variable> found - either without "
+        "'id' or 'datatype'") in str(e)
+
+
+def test_check_and_correct_xccdf_to_oval_data_export_matching_constraints_2(
+        xccdf_with_value, oval_with_external_variable_without_id):
+    with pytest.raises(SSGError) as e:
+        ssg.build_renumber.check_and_correct_xccdf_to_oval_data_export_matching_constraints(
+            xccdf_with_value, oval_with_external_variable_without_id)
+    assert (
+        "Invalid OVAL <external_variable> found - either without "
+        "'id' or 'datatype'") in str(e)
+
+
+def test_check_and_correct_xccdf_to_oval_data_export_matching_constraints_3(
+        xccdf_with_value, oval_with_external_variable_without_datatype):
+    with pytest.raises(SSGError) as e:
+        ssg.build_renumber.check_and_correct_xccdf_to_oval_data_export_matching_constraints(
+            xccdf_with_value, oval_with_external_variable_without_datatype)
+    assert (
+        "Invalid OVAL <external_variable> found - either without "
+        "'id' or 'datatype'") in str(e)
+
+
+def test_check_and_correct_xccdf_to_oval_data_export_matching_constraints_4(
+        xccdf_with_incomplete_value, oval_with_external_variable):
+    with pytest.raises(SSGError) as e:
+        ssg.build_renumber.check_and_correct_xccdf_to_oval_data_export_matching_constraints(
+            xccdf_with_incomplete_value, oval_with_external_variable)
+    assert (
+        "Invalid XCCDF variable 'var_password_age': Missing the"
+        " 'type' attribute") in str(e)
+
+
+def test_check_and_correct_xccdf_to_oval_data_export_matching_constraints_5(
+        xccdf_with_value, oval_with_external_variable):
+    ssg.build_renumber.check_and_correct_xccdf_to_oval_data_export_matching_constraints(
+        xccdf_with_value, oval_with_external_variable)
+    assert xccdf_with_value is not None
+
+
+@pytest.fixture
+def xccdf_with_invalid_cce():
```

Comment:
Would it be possible to add a test of `verify_correct_form_of_referenced_cce_identifiers` with a rule without a CCE?